submissions/Mifaresss/HTML-CSS-Popup/css/styles.css

```diff
@@ -0,0 +1,223 @@
+/* ========== Common Styles ========== */
+body {
+	font-family: 'Noto Sans', sans-serif;
+	font-size: 20px;
+	line-height: 1.5;
+	overflow-x: hidden;
+}
+* {
+	box-sizing: border-box;
+}
+*::-webkit-scrollbar {
+	width: 15px;
+}
+*::-webkit-scrollbar-track {
+	background-color: rgba(170, 170, 170, 0.644);
+}
+*::-webkit-scrollbar-thumb {
+	background: linear-gradient(to bottom, #cfecd0, #ffc5ca);
+	border-radius: 5px;
+}
+
+
+
+/* ========== Header ========== */
+.header {
+	padding: 20px 15px 50px 15px;
+	height: 700px;
+}
+.wrapper {
+	display: flex;
+	padding: 0 15px;
+}
+.header__container {
+	margin-right: 45px;
+	margin-left: auto;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	gap: 30px;
+}
+.header__logo {
+	font-size: 46px;
+	color: #212121;
+	font-weight: 700;
+	transition: all .3s;
+}
+.header__logo:hover {
+	color: #1b4903;
+	transform: scale(1.05,1.05);
+}
+.header__logo:focus {
+	border: 2px solid #1b4903;
+	border-radius: 5px;
+	color: #1b4903;
+	transform: scale(1.05,1.05);
+}
+.wrapper-popup {
+	position: relative;
+	width: 35px;
+	height: 35px;
+}
+.label {
+	cursor: pointer;
+	transition: all .3s;
+}
+.label svg {
+	transition: all .4s;
+}
+.label:hover svg {
+	transform: scale(1.05,1.05);
+}
+.label:hover svg path {
+	fill: #1b4903;
+}
+.popup-button {
+	position: absolute;
+	appearance: none;
+}
+.popup-button:focus {
+	border: 2px solid #1b4903;
+	border-radius: 5px;
+	padding: 25px;
+	top: -10px;
+	left: -9px;
+	cursor: pointer;
+}
+.popup {
+	position: absolute;
+	top: -1000px;
+	right: -185px;
+	transition: all 0.8s;
+	height: 470px;
+	overflow-y: scroll;
+	overflow-x: hidden;
+	z-index: 100;
+	border: 2px solid #212121;
+	border-radius: 5px;
+	padding: 20px 35px;
+}
+.popup-button:checked ~ .popup {
+	top: 75px;
+	background-color: #fff;
+	box-shadow: 0px 0px 8px #212121;
+}
+.popup__list {
+	display: grid;
+	grid-template-columns: 1fr 1fr 1fr;
+	grid-template-rows: 1fr 1fr;
+	gap: 40px;
+	align-items: center;
+	justify-content: center;
+}
+.popup__link {
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+	font-size: 16px;
+	color: #212121;
+	justify-content: center;
+	gap: 9px;
+}
+.popup__link img {
```

Comment:
Avoid using too specific CSS selectors. [Read more](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity)

Reply:
I don't understand what kind of selectors you are talking about. Here only simple classes are used. Please clarify

Reply:
The current CSS selector consists of two elements block with class `popup_link` and `img` tag. So you have one level of nesting. 
How browser parsed CSS selectors: from right to left. 
In the current case, the browser creates a collection with all `img` elements on the page he filtered that collection and left only `img` elements with the parent block with class`popup_link`. So the more nesting levels in your code, the greater the load on the browser and the slower your styles will be applied. Also, if you need to reassign some value in that selector you should, at least, repeat that CSS selector to get success.
As result: minimize the specificity of your CSS selectors. and avoid styling by HTML tags.

Reply:
Ok, thx for the explanation